src/main/scala/loamstream/apps/IndexFiles.scala

```diff
@@ -0,0 +1,69 @@
+package loamstream.apps
+
+import java.io.FileWriter
+import java.nio.file.Path
+
+import org.apache.commons.csv.CSVFormat
+import org.apache.commons.csv.CSVPrinter
+
+import loamstream.conf.ExecutionConfig
+import loamstream.model.jobs.Execution
+import loamstream.model.jobs.LJob
+import loamstream.util.CanBeClosed
+import loamstream.util.Files
+
+/**
+ * @author clint
+ * Jun 4, 2019
+ */
+object IndexFiles {
+  def writeIndexFiles(
+    executionConfig: ExecutionConfig,
+    jobsToExecutions: Map[LJob, Execution]): Unit = {
+
+    def makePath(fileName: String) = executionConfig.logDir.resolve(fileName)
+
+    Files.createDirsIfNecessary(executionConfig.logDir)
+    
+    writeDataTo(makePath("all-jobs.tsv"))(jobsToExecutions)
+
+    import loamstream.util.Maps.Implicits._
+    
+    writeDataTo(makePath("failed-jobs.tsv"))(jobsToExecutions.filterValues(_.isFailure))
+  }
+  
+  private def writeDataTo(file: Path)(executionTuples: Iterable[(LJob, Execution)]): Unit = {
+    val csvFormat = CSVFormat.DEFAULT.
+        withHeader("JOB_ID", "JOB_NAME", "JOB_STATUS", "EXIT_CODE", "JOB_DIR").
+        withDelimiter('\t').
+        withRecordSeparator('\n')
+    
+    val toRow = Row.tupled
+    
+    val sorted = executionTuples.toSeq.sortWith(Orderings.executionTupleOrdering)
+    
+    CanBeClosed.enclosed(new CSVPrinter(new FileWriter(file.toFile), csvFormat)) { csvPrinter =>
+      executionTuples.iterator.map(toRow).map(_.toJavaIterable).foreach(csvPrinter.printRecord)
+    }
+  }
+  
+  private final case class Row(job: LJob, ex: Execution) {
+    def toJavaIterable: java.lang.Iterable[String] = {
+      import scala.collection.JavaConverters._
+      
+      Iterable(job.id.toString, job.name, jobStatusPart(ex), exitCodePart(ex), jobDirPart(ex)).asJava
```

Comment:
While I'm usually one for doing the simplest thing, this is one case where I'd suggest actually going a bit more complex, making a `case class` for the definition of a row, and then using `productIterator` to ensure that all the headers/columns are in the right order and you didn't miss anything:

```
def makeCSVFormat[T <: Product](): CSVFormat = {
  CSVFormat.DEFAULT.withHeaders(classOf[T].getDeclaredFields.map(_.getName))
}

case class JobRow(id: String, name: String, ...) {
  def csvRow() = productIterator.to.asJava
}
```

I've just done this in the past where I added a column or re-organized and forgot to update later, and since every column is the same type, it "just worked" but all the data was wrong because the columns didn't match w/ the headers, etc.

Reply:
Your concerns are absolutely well-founded.  Unfortunately, making this work is a little clunky: `productIterator` returns values of fields in declaration order, and 
```scala
import scala.reflect.runtime.universe._
def fieldNames[A](implicit classTag: TypeTag[A]): Seq[String] = {
  typeOf[A].members.collect {
    case m: MethodSymbol if m.isCaseAccessor => m.name.toString
  }
}
```
returns the field names in _reverse_ declaration order, or at least, that's what it appears like - I have no idea if that's something I can rely on.  The order of elements produced by `productIterator` appears to be intentional and guaranteed (and in line with `productElement`), but I can't find anything about how member symbols are ordered by `typeOf[A].members`, or if any guarantees exist at all there.

So I'm going to skip this one.  You're right that it's lame and asking for trouble to hard-code column names and positions, but I'd rather deal with errors caused by that (for now) instead of ones caused by some arbitrarily-shifted field-name ordering, since the former will likely be easier to fix.

This is a small, human-readable file that's a summary of information available elsewhere, so I think the risk is low enough to justify leaving this as-is.